### ISSUE TITLE
names for function-style group-by keys

### ISSUE THREAD

**Issue by mccanne** [brimsec/zq#1378]
The zql grammar currently requires that group-by keys specified as a function include a key name using syntax like this:
```
count() by f1=foo(),f2=bar()
```
where the key names are `f1` and `f2`.

We should also by able to say :
```
count() by foo(),bar()
```
and the implied keys would be `foo` and `bar`.

Of course, this will still be an error because it's not clear how to name it:
```
count() by foo()+1
```

**Comment by henridf** [comment]
foo() and bar() imply easy names as they have no parameters. That said, none of our current functions are nullary.  

What about `count() by String.toUpper(host), String.toUpper(name)` ?

**Comment by philrz** [comment]
@henridf: There's already code that errors on repeated field names when the key names are explicitly specified, e.g.:

```
$ zq -f table 'len(answers) > 0 | count() by f1=len(answers),f1=len(answers)' dns.log.gz 
compiling groupby: field f1 is repeated
```

Could we just use the same logic to kick back an error in your `toUpper` example?

**Comment by henridf** [comment]
@philrz yes implementing this should be easy.

**Comment by mccanne** [comment]
I think this is done no?
@philrz ?

**Comment by philrz** [comment]
@mccanne: Indeed! I just did a binary search and confirmed this got addressed way back in October as part of #1468.

Verifying with current `zq` at commit `b3dc877e`, let's start with the original situation where we needed the key names.

```
$ echo '{"fruit": "apple", "color": "RED", "price": 1.99}' | zq -f zson 'count() by bigfruit=to_upper(fruit),smallcolor=to_lower(color),roundprice=ceil(price)' -
{
    bigfruit: "APPLE",
    smallcolor: "red",
    roundprice: 2e+00,
    count: 1 (uint64)
} (=0)
```

We can drop the explicit key names and have them named after the functions we called.

```
$ echo '{"fruit": "apple", "color": "RED", "price": 1.99}' | zq -f zson 'count() by to_upper(fruit),to_lower(color),ceil(price)' -
{
    to_upper: "APPLE",
    to_lower: "red",
    ceil: 2e+00,
    count: 1 (uint64)
} (=0)
```

And as @mccanne mentioned, if we do more with the expression beyond just invoking the function, it's an error.

```
$ echo '{"fruit": "apple", "color": "RED", "price": 1.99}' | zq -f zson 'count() by to_upper(fruit),to_lower(color),ceil(price)+1' -
assignment name could not be inferred from rhs expressioin
```

Thanks @mccanne!
